channeldb/mp_payment.go

```diff
@@ -8,26 +8,32 @@ import (
 	"github.com/lightningnetwork/lnd/routing/route"
 )
 
-// HTLCAttemptInfo contains static information about a specific HTLC attempt
+// HTLCWireInfo contains static information about a specific HTLC attempt
 // for a payment. This information is used by the router to handle any errors
 // coming back after an attempt is made, and to query the switch about the
 // status of the attempt.
-type HTLCAttemptInfo struct {
-	// AttemptID is the unique ID used for this attempt.
-	AttemptID uint64
-
+type HTLCWireInfo struct {
 	// SessionKey is the ephemeral key used for this attempt.
 	SessionKey *btcec.PrivateKey
 
 	// Route is the route attempted to send the HTLC.
 	Route route.Route
 }
 
+// HTLCAttemptInfo contains the wire info for the htlc and a unique identifier
+// for this attempt.
+type HTLCAttemptInfo struct {
```

Comment:
why do we have to split this struct into two?

Reply:
Commit dropped. The idea was to make the mapping of this struct to serialize/deserialize methods complete. Now the payment id field is populated separately, but no big deal.